polygeist/tools/cgeist/Test/Verification/sycl/restrict.cpp

```diff
@@ -3,25 +3,44 @@
+// CHECK-MLIR-DAG: func.func @test_struct(%arg0: !llvm.ptr<struct<(i32)>, 4> {{{.*}}llvm.noalias{{.*}}}, %arg1: !llvm.ptr<struct<(i32)>, 4> {{{.*}}llvm.noalias{{.*}}})
+// CHECK-LLVM-DAG: define spir_func void @test_struct({ i32 } addrspace(4)* noalias {{.*}}%0, { i32 } addrspace(4)* noalias {{.*}}%1)
 struct S {
   int i;
 };
 extern "C" SYCL_EXTERNAL void test_struct(struct S * __restrict__ a, struct S * __restrict__ b) {}
 
-// CHECK-MLIR: func.func @test_vec(%arg0: memref<?x!sycl_vec_f64_16_, 4> {{{.*}}llvm.noalias{{.*}}}, %arg1: memref<?x!sycl_vec_f64_16_, 4> {{{.*}}llvm.noalias{{.*}}})
-// CHECK-LLVM: define spir_func void @test_vec(%"class.sycl::_V1::vec" addrspace(4)* noalias {{.*}}%0, %"class.sycl::_V1::vec" addrspace(4)* noalias {{.*}}%1)
+// CHECK-MLIR-DAG: func.func @test_vec(%arg0: memref<?x!sycl_vec_f64_16_, 4> {{{.*}}llvm.noalias{{.*}}}, %arg1: memref<?x!sycl_vec_f64_16_, 4> {{{.*}}llvm.noalias{{.*}}})
+// CHECK-LLVM-DAG: define spir_func void @test_vec(%"class.sycl::_V1::vec" addrspace(4)* noalias {{.*}}%0, %"class.sycl::_V1::vec" addrspace(4)* noalias {{.*}}%1)
 extern "C" SYCL_EXTERNAL void test_vec(sycl::vec<sycl::cl_double, 16> * __restrict__ a, const sycl::vec<sycl::cl_double, 16> * __restrict__ b) {}
 
-// CHECK-MLIR: func.func @_ZN1B10test_classEP1APS_(%arg0: !llvm.ptr<struct<(i8)>, 4> {{.*}}, %arg1: !llvm.ptr<struct<(i8)>, 4> {llvm.noalias{{.*}}}, %arg2: !llvm.ptr<struct<(i8)>, 4> {llvm.noalias{{.*}}})
-// CHECK-LLVM: define linkonce_odr spir_func void @_ZN1B10test_classEP1APS_({ i8 } addrspace(4)* {{.*}}%0, { i8 } addrspace(4)* noalias {{.*}}%1, { i8 } addrspace(4)* noalias {{.*}}%2)
+// CHECK-MLIR-DAG: func.func @_ZN1B10test_classEP1APS_(%arg0: !llvm.ptr<struct<(i8)>, 4> {{.*}}, %arg1: !llvm.ptr<struct<(i8)>, 4> {llvm.noalias{{.*}}}, %arg2: !llvm.ptr<struct<(i8)>, 4> {llvm.noalias{{.*}}})
+// CHECK-LLVM-DAG: define linkonce_odr spir_func void @_ZN1B10test_classEP1APS_({ i8 } addrspace(4)* {{.*}}%0, { i8 } addrspace(4)* noalias {{.*}}%1, { i8 } addrspace(4)* noalias {{.*}}%2)
 class A {};
 class B {
   SYCL_EXTERNAL void test_class(class A * __restrict__ a, class B * __restrict__ b) {}
 };
+
+// CHECK-MLIR-DAG: gpu.func @{{.*}}kernel_args_restrict(%arg0: memref<?xi32, 1> {{{.*}}llvm.noalias{{.*}}}, %arg1: memref<?x!sycl_range_1_>{{.*}}, %arg2: memref<?x!sycl_range_1_>{{.*}}, %arg3: memref<?x!sycl_id_1_>{{.*}})
+// CHECK-LLVM-DAG: define weak_odr spir_kernel void @{{.*}}kernel_args_restrict(i32 addrspace(1)* {{.*}}noalias{{.*}} %0, %"class.sycl::_V1::range.1"* {{.*}} %1, %"class.sycl::_V1::range.1"* {{.*}} %2, %"class.sycl::_V1::id.1"* {{.*}} %3)
+using namespace sycl;
+int args_restrict(std::array<int, 1> &A) {
+  queue q;
+  {
+    auto buf = buffer<int, 1>{A.data(), 1};
+    q.submit([&](handler &cgh) {
+      auto A = buf.get_access<access::mode::write>(cgh);
+      cgh.single_task<class kernel_args_restrict>(
+          [=]() [[intel::kernel_args_restrict]]{
+            A[0] = 1;
```

Comment:
[nit]: I see that the argument has the `llvm.noalias` qualifier. The test would be more meaningful if it has 2 arguments both qualified with that attribute.

Reply:
done.